### ISSUE TITLE
SDC: MediaView should be non-mandatory for custom components

### ISSUE THREAD

**Issue by khyativyasargus** [google/android-fhir#2002]
**Describe the bug**
Currently when we create custom ui component with parent class QuestionnaireItemViewHolderFactory, MediaView(View from SDC) is mandatory to implement in custom xml, as it has usage in parent class.

**Component**
SDC library

**To Reproduce**
1. Create Custom component mentioned in [documentation](https://github.com/google/android-fhir/wiki/SDCL%3A-Customize-how-a-Questionnaire-is-displayed#create-a-custom-component) with parent class QuestionnaireItemViewHolderFactory
2. Create xml for your component and implement all other steps of mapping
3. You would notice , MediaView is compulsory to add in your custom layout xml as It's reference is being used in QuestionnaireItemViewHolderFactory and it doesn't have any condition to handle it's availability.

**Expected behavior**
Usage of MediaView should be handled according to its availability as custom components won't necessarily have need of MediaView in their custom view

**Screenshots**

@jingtang10 @FikriMilano

**Comment by FikriMilano** [comment]
@khyativyasargus have you tried to set the MediaView XML visibility to gone? And see if that works for you

**Comment by khyativyasargus** [comment]
> @khyativyasargus have you tried to set the MediaView XML visibility to gone? And see if that works for you

Yes, currently that's what we have done for now. But that isn't an ideal way to do right? We shouldn't force application users to add MediaView compulsory on their custom XML.

**Comment by FikriMilano** [comment]
Yeah.. I can agree, adding item-media extension itself to questionnaire.item is optional right?

We put the MediaView to the parent layout itself is to reduce code repetition in the code. But now, your argument does makes sense, so thank you.

@jingtang10 let me know what you think.
